File: backend/src/main/java/com/decmoe47/todo/service/UserService.java
package com.decmoe47.todo.service;

import com.decmoe47.todo.model.dto.UserSearchDTO;
import com.decmoe47.todo.model.dto.UserUpdateDTO;
import com.decmoe47.todo.model.vo.UserVO;
import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.List;

public interface UserService extends UserDetailsService {

    UserVO getUser(long userId);

    List<UserVO> searchUser(UserSearchDTO searchUserDTO);

    UserVO updateUser(long userId, UserUpdateDTO userUpdateDTO);

    UserVO getUserByToken(String token);
}

File: backend/src/main/kotlin/com/decmoe47/todo/service/UserService.kt
package com.decmoe47.todo.service

import com.decmoe47.todo.model.request.UserSearchRequest
import com.decmoe47.todo.model.request.UserUpdateRequest
import com.decmoe47.todo.model.response.UserResponse
import org.springframework.security.core.userdetails.UserDetailsService

interface UserService : UserDetailsService {
    fun getUser(userId: Long): UserResponse
    fun searchUser(request: UserSearchRequest): List<UserResponse>
    fun updateUser(userId: Long, request: UserUpdateRequest): UserResponse
    fun getUserByToken(token: String): UserResponse
}
